java/src/main/java/com/genexus/reports/PDFReportPDFBox.java

```diff
@@ -637,9 +639,24 @@ else if (valign == PDFReportPDFBox.VerticalAlign.BOTTOM.value())
 
 					Document document = Jsoup.parse(sTxt);
 					Elements allElements = document.getAllElements();
-					for (Element element : allElements)
+					for (Element element : allElements){
+						if (pageHeightExceeded(bottomMargin, spaceHandler.getCurrentYPosition())) {
+							llx = leftAux + leftMargin;
+							lly = drawingPageHeight - bottomAux;
+							urx = rightAux + leftMargin;
+							ury = drawingPageHeight - topAux;
+							htmlRectangle = new PDRectangle(llx, lly, urx - llx, ury - lly);
+							spaceHandler = new SpaceHandler(htmlRectangle.getUpperRightY(), htmlRectangle.getHeight());
+							bottomAux -= drawingPageHeight;
+							GxEndPage();
+							GxStartPage();
+
+							cb.close();
+							cb = new PDPageContentStream(this.document, this.document.getPage(page - 1),PDPageContentStream.AppendMode.APPEND,false);
```

Comment:
Why do you use this.document instead of only document?
This line it seems to be the same that is in line 576.
May be it is good to write it in a method so if any time it's changed you will be sure to change it in both side.

Reply:
`Document document = Jsoup.parse(sTxt);` is previously defined on line 640 so I need the "this" keyword to reference the attribute and not the variable. Do you think it better to get rid of the "this" keyword and change the name of the variable on line 640?

Reply:
Uhhh, OK
May be it will be better to use another name for clarification